src/sensors.py

```diff
@@ -315,3 +372,29 @@ def setup_sensors(instrument, katcp_server):
         katcp_server.add_sensor(sensor)
         instrument._sensors[sensor.name] = sensor
         ioloop.add_callback(_feng_pfb_okay, sensor, executor, _f)
+
+    # f-engine comms rx
+    fengops = corr2.fxcorrelator_fengops.FEngineOperations(instrument)
+    sensor = Sensor(sensor_type=Sensor.BOOLEAN, name='fhosts_check_rx',
+                    description='F-hosts rx okay',
+                    default=True)
+    katcp_server.add_sensor(sensor)
+    instrument._sensors[sensor.name] = sensor
+    ioloop.add_callback(_fhost_check_rx, sensor, executor, fengops)
+
+    # # f-engine comms tx
+    sensor = Sensor(sensor_type=Sensor.BOOLEAN, name='fhosts_check_tx',
+                    description='F-hosts tx okay',
+                    default=True)
+    katcp_server.add_sensor(sensor)
+    instrument._sensors[sensor.name] = sensor
+    ioloop.add_callback(_fhost_check_tx, sensor, executor, fengops)
+
+    # x-engine comms rx
+    xengops = corr2.fxcorrelator_xengops.XEngineOperations(instrument)
```

Comment:
Could you not replace this with

xengops = instrument.xops 

or something like that?

@paulprozesky Am I on the right track?